src/node.cc

```diff
@@ -3403,6 +3409,14 @@ static void ParseArgs(int* argc,
       printf("%s\n", NODE_VERSION);
       exit(0);
     } else if (strcmp(arg, "--help") == 0 || strcmp(arg, "-h") == 0) {
+      if (strcmp(NODE_PLATFORM, "win") != 0 &&
+          strcmp(NODE_PLATFORM, "win32") != 0) {
```

Comment:
It won't cause a regression for them, they'll just get the regular old help, and that's fine, IMO.  You shouldn't need to check for "win", by the way, it should always be "win32" on Windows.